{"version":1,"ops":[{"type":1,"author":{"id":"2182ad57f6c7d53407dc493295dd8351df3186d6"},"timestamp":1408616896,"metadata":{"github-id":"MDU6SXNzdWU0MDc5MTI1Mg==","github-url":"https://github.com/DFHack/dfhack/issues/293","origin":"github"},"title":"40.8-r2: showmood crashes","message":"I managed to crash the game by calling showmood, in a rather special situation. I think it's a DF bug, meaning the situation shouldn't occur and showmood crashes because it doesn't expect it...\n\nHere's a savegame: https://dl.dropboxusercontent.com/u/90977204/showmood_crash.7z\n\nSituation is as follows: \nA dwarf fell into a mood and started crafting (craftsdwarf shop is still claimed, right left to the volcano on the surface).\nAn ambush came, and the moody dwarf escaped to the edge of the map (I think, I didn't actually watch him), where he seems to be stuck now; near the top left corner of the map.\nCalling showmood crashes the game, for the following reason:\n\"Unhandled exception at 0x684f1854 (showmood.plug.dll) in Dwarf Fortress.exe: 0xC0000005: Access violation reading location 0xabababab.\"\n\nExamining this in the debugger shows it occurs in showmood.cpp, line 173:\ndf::item_type type = job-\u003ejob_items[i]-\u003eitem_type;\n\nThis is a loop running from 0 to job-\u003eitems.size()-1. But the problem is, the vectors job-\u003eitems and job-\u003ejob_items   have different sizes, job-\u003eitems has 2 entries, job-\u003ejob_items only 1. So it's trying to access memory out of bounds.","files":null}]}